lang/py/src/avro/schema.py

```diff
@@ -323,6 +323,21 @@ class LogicalSchema(object):
   def __init__(self, logical_type):
     self.logical_type = logical_type
 
+# Base class for time related logical types
+class _TimeSchema(LogicalSchema):
+  def __init__(self, logical_type, fsp):
+    max_fsp = self.max_fsp()
+    if not isinstance(fsp, int):
+      raise SchemaParseException('Fsp (Integer) is required for logical type %s.' % logical_type)
+
+    if fsp > max_fsp:
```

Comment:
This should not necessary. It's known and controlled by the derived class, not by users.